test/integration/acceptance/remove_peer_test.cpp

```diff
@@ -0,0 +1,100 @@
+                            interface::types::PubkeyType pubkey) {
+  return ::testing::Truly(
+      [address = std::move(address),
+       pubkey = std::move(pubkey)](std::shared_ptr<interface::Peer> peer) {
+        return peer->address() == address and peer->pubkey() == pubkey;
+      });
+}
+
+auto makePeerPointeeMatcher(std::shared_ptr<interface::Peer> peer) {
+  return makePeerPointeeMatcher(peer->address(), peer->pubkey());
+}
+
+/**
+ * @given a network of one real and one fake peers
+ * @when fake peer is removed
+ * @then the transaction is committed
+ *    @and the ledger state after commit contains one peer,
+ *    @and the WSV reports that there is one peer
+ */
+TEST_F(FakePeerExampleFixture, FakePeerIsRemoved) {
+  // ------------------------ GIVEN ------------------------
+  // init the real peer with one fake peer in the genesis block
+  createFakePeers(1);
+  auto &itf = prepareState();
+  const auto prepared_height = itf.getBlockQuery()->getTopBlockHeight();
+  auto fake_peer = fake_peers_.front();
+
+  // capture itf synchronization events
+  auto itf_sync_events_observable = itf_->getPcsOnCommitObservable().replay();
+  itf_sync_events_observable.connect();
+
+  // ------------------------ WHEN -------------------------
+  // send removePeer command
+  itf.sendTxAwait(
+      complete(baseTx(kAdminId).removePeer(fake_peer->getAddress(),
+                                           fake_peer->getKeypair().publicKey()),
+               kAdminKeypair),
+      checkBlockHasNTxs<1>);
+
+  // ------------------------ THEN -------------------------
+  // check that ledger state contains one peer
+  itf_sync_events_observable
+      .filter([prepared_height](const auto &sync_event) {
+        return sync_event.ledger_state->top_block_info.height > prepared_height;
+      })
+      .take(1)
+      .timeout(kSynchronizerWaitingTime, rxcpp::observe_on_new_thread())
+      .as_blocking()
+      .subscribe(
+          [&, itf_peer = itf_->getThisPeer()](const auto &sync_event) {
+            EXPECT_THAT(sync_event.ledger_state->ledger_peers,
+                        ::testing::UnorderedElementsAre(
+                            makePeerPointeeMatcher(itf_peer)));
+          },
+          [](std::exception_ptr ep) {
+            try {
+              std::rethrow_exception(ep);
+            } catch (const std::exception &e) {
+              FAIL() << "Error waiting for synchronization: " << e.what();
+            }
+          });
+
+  // query WSV peers
+  auto opt_peers = itf.getIrohaInstance()
+                       .getIrohaInstance()
+                       ->getStorage()
+                       ->createPeerQuery()
+                       .value()
+                       ->getLedgerPeers();
+
+  // check the two peers are there
+  ASSERT_TRUE(opt_peers);
+  EXPECT_THAT(*opt_peers,
+              ::testing::UnorderedElementsAre(
+                  makePeerPointeeMatcher(itf.getThisPeer())));
+}
```

Comment:
there could also be a test where the real peer gets removed. it could check that it terminates gracefully (i.e. with no segfaults and uncaught exceptions).